src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java

```diff
@@ -191,7 +191,7 @@ private void addPieceIfValid(Piece piece, Map<String, List<Piece>> piecesByPoLin
     // request
     if (piecesByLineId.containsKey(poLineId) && piecesByLineId.get(poLineId).containsKey(pieceId)) {
       // Validate if the piece is not yet received
-      if (piece.getReceivingStatus() == ReceivingStatus.EXPECTED || isRevertToOnOrder(piece)) {
+      if (!ReceivingStatus.RECEIVED.equals(piece.getReceivingStatus()) || isRevertToOnOrder(piece)) {
```

Comment:
**Minor**
Please checked enum values with `==` and `!=`. I think it's natural to have `piece.getReceivingStatus()` on the left side of operator as we are checking it's value.

Reply:
Done. Thank you!